### PR TITLE
Standardize Naming Conventions for Query-Related Services in `app/queries`

### PR DESCRIPTION
**Pull request by feruzoripov**
#### What? Why?

In this pull request, I standardized the naming conventions for services related to queries in the `app/queries` folder. Previously, some of these services ended with the `...Query` postfix, while others did not. To enhance clarity and maintain consistency, I renamed all relevant class names to end with `...Query`. This standardization makes it easier for developers to identify these services as query-related and avoid potential misunderstandings about their purpose. Additionally, I updated the corresponding code throughout the codebase to reflect these changes. Also, I renamed methods named `query` to `call`.

#### What should we test?

- make sure tests are passing since it does not change business logic.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

#### Documentation updates

For future changes, please adhere to this naming convention when introducing new query-related services in the `app/queries` folder. This consistency will streamline code readability and maintain a clear understanding of the purpose of each service.

### CODE REVIEW

**Review by rioug — APPROVED** [review]
Thanks for your help on this :pray: ! I left a few comment, but I think they have been addressed later in the cleanup/cop commits. 
The commit history is a bit hard to follow at the end, ideally the clean up commit should have been squashed.

File: spec/queries/customers_with_balance_query_spec.rb [hunk 1]
```diff
@@ -2,97 +2,100 @@
...
         end
       end
     end
 
     context 'when orders are in cart state' do
       before do
-        create(:order, customer:, total: order_total, payment_total: 0, state: 'cart')
-        create(:order, customer:, total: order_total, payment_total: 0, state: 'cart')
+        create(:order, customer: customers, total: order_total, payment_total: 0, state: 'cart')
+        create(:order, customer: customers, total: order_total, payment_total: 0, state: 'cart')
```
Comment (dacook) [inline spec/queries/customers_with_balance_query_spec.rb:43]: At first the `let(:customers)` refactor looked like a good idea, but now we're using it in different contexts, and here it's confusing because the name is plural and there's only one. It affects a number of specs below, so I'm not sure it makes the code better.

I can't think of a better way to arrange it right now, so would suggest reverting that change, would that be ok?

**Review by dacook — CHANGES_REQUESTED** [review]
Thank you for this contribution, it really helps. The strength of Rails is that everything is arranged in a pre-determined way.

It would be great if there is a webpage you can refer to that documents this standard, and we could add it to our wiki. 

I'm curious if the `call` method is necessary, or if we can go further and run the query in the initialiser instead. But maybe that's going too far ;) If there's a standard, we should stick to it.

Regarding the `customers` variable in specs, I suggest we keep it as it was. Would it be hard to undo?

File: spec/queries/customers_with_balance_query_spec.rb (hunk 1 above)
Comment (feruzoripov) [reply spec/queries/customers_with_balance_query_spec.rb]: @dacook thank you for your feedback. While there may not be official documentation explicitly stating these conventions, adhering to such practices is widely recognized within the development community making the codebase more readable and maintainable.

Regarding the `call` method, its use aligns with common design patterns in Ruby on Rails, providing a clear interface for executing main functionalities. Running the query in the initializer might be possible, but it could potentially lead to unexpected behavior, and it may not align with the typical design patterns in Rails. So I wouldn't suggest running any logic in the initializer :)

As for the `customers` variable in specs, I updated it to not affect `order` creation in specs below. cc: @rioug

**Review by dacook — APPROVED** [review]
Thanks so much for the follow up, this is a solid contribution.

Good point about the `call` method performing the main function 👍

**Comment by dacook** [comment]
I've updated documentation in the wiki here: https://github.com/openfoodfoundation/openfoodnetwork/wiki/Code-Conventions

I believe these changes are safe to not require any further testing. Merging! 🚀
